test/e2e_node/jenkins/ubuntu-14.04-nvidia-install.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is meant to install Nvidia drivers on Ubuntu 14.04 GCE VMs.
+# This script is meant to facilitate testing of Nvidia GPU support in Kubernetes.
+echo "Checking for CUDA and installing."
+# Check for CUDA and try to install.
+if ! dpkg-query -W cuda; then
+  curl -O http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1404/x86_64/cuda-repo-ubuntu1404_8.0.61-1_amd64.deb
+  dpkg -i ./cuda-repo-ubuntu1404_8.0.61-1_amd64.deb
+  apt-get update
+  apt-get install cuda -y
+  apt-get install linux-headers-$(uname -r) -y
+fi
+## Pre-loads kernel modules
+nvidia-smi
```

Comment:
Maybe worth double checking whether this happens as part of `modprobe` or if something else needs to do it out of band (running `nvidia-smi` as root works, but I don't think it's just because it loads the modules).

Slightly older versions of nvidia's Linux getting-started guides recommend (for systems where applications can't use `nvidia-modprobe` due to setuid restrictions) running (as root at boot) `nvidia-smi` or running a shell script that `modprobe`s each module and then manually does `mknod`. Newer versions of the guides are just recommending the shell script, which makes me wonder why they stopped recommending `nvidia-smi`.
Older: http://developer.download.nvidia.com/compute/cuda/7_0/Prod/doc/CUDA_Getting_Started_Linux.pdf
Newer: http://developer.download.nvidia.com/compute/cuda/7.5/Prod/docs/sidebar/CUDA_Installation_Guide_Linux.pdf
Grep for `4.4. Device Node Verification` in either of those.

I know, for example, on my home setup, where I use NixOS, I don't have `nvidia-modprobe` but the nix configuration for the nvidia drivers is set up to create the node for nvidia-uvm: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/hardware/video/nvidia.nix#L69

Granted, you've probably already seen all this so if that's the case then no worries. I still think it may be worth noting the device file creation in a comment.